### PR TITLE
fix(web): reject hand-assembled callsite stacks

### PR DESCRIPTION
**Pull request by jan-kubica**
## Proposed change

Follow-up to #2744, which retains SpiderMonkey and JavaScriptCore frames when redacting telemetry stacks.

Neither callsite engine writes a `<name>: <message>` header, so a callsite stack whose first non-empty line is not a frame was assembled from text (a copied message, a rehydrated V8 stack). Callsite-shaped lines under such a line cannot be told apart from message content. `redactTelemetryStack` now returns `undefined` for callsite syntax unless the stack opens with a callsite frame or a JavaScriptCore `<symbol>@[native code]` built-in frame; V8 handling is unchanged.

The workflow test for the Firefox/WebKit guard now parses the ci-plan selector and asserts set equality with the complete owner list in both directions, so a dropped owner fails the test instead of silently skipping the job.

## Tests

- Unit: a message line above a callsite-shaped line yields no stack; a frame-shaped header (`TypeError: notify jane@https://…:1:2`) yields no stack under both syntaxes; frames under a native built-in frame survive. Disabling the guard fails the first test.
- Browser harness (native Firefox and WebKit): prepending message text to a live stack yields no redacted stack; the existing frame-retention case still passes.
- Workflow: removing an owner from the ci.yml selector fails the census test.

https://claude.ai/code/session_01ScmEtmrqPZUhzmtxFRZgEt

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved telemetry stack redaction across Firefox, WebKit and Chromium.
  - Invalid or manually assembled error stacks are now discarded instead of being submitted as misleading data.
  - Valid callsite frames continue to be retained, including stacks containing native browser frames.

- **Tests**
  - Added coverage for browser-specific stack formats and edge cases.
  - Expanded safeguards to ensure all relevant stack-redaction changes trigger the appropriate cross-browser checks.

### CODE REVIEW

File: apps/web/src/lib/analytics/posthog.ts [hunk 1]
```diff
@@ -107,15 +107,30 @@ const deepestCause = (error: Error): Error => {
...
+const GECKO_FRAME = /^[^@]*@\S+:\/\/\S+:\d+:\d+$/u;
+
+// The message can carry PII and lives only on V8's header line, so the
+// syntax is chosen per stack rather than per line: a stack holding V8 frames
+// is read as V8 only, and no header can satisfy the other pattern's frame
+// shape. A header-free non-V8 stack has no message to leak.
+const frameLines = (stack: string): string[] => {
+  const lines = stack.split("\n");
+  const v8 = lines.filter((line) => V8_FRAME.test(line));
+  return v8.length > 0 ? v8 : lines.filter((line) => GECKO_FRAME.test(line));
```
[the remark this answers is not in the document]
Comment (jan-kubica) [reply apps/web/src/lib/analytics/posthog.ts]: Addressed in 3bcfe451436ab2befd35778a2b24237d14ff8818. The production parser from this branch was replaced by the one merged in #2744, where V8 syntax keeps only `at` lines and a header is never a frame. This PR pins the header-only invariant engine-independently: `TypeError: notify jane@https://my.stll.app/matters/Client-Smith:1:2` with no frames yields no stack under both V8 and callsite syntax, and a callsite stack must open with a frame at all.

CC on behalf of jan-kubica

**Comment by jan-kubica** [comment]
Head amended to 3b022f670fe27da173a218dd8a3bc1106edff3b1: lint-only change in the census test (`RegExp#exec` instead of `String#match`), no behavior change from 3bcfe45143.

CC on behalf of jan-kubica
